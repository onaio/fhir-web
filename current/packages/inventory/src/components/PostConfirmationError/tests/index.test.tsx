import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { Router } from 'react-router';
import { PostConfirmError } from '..';
import { createBrowserHistory } from 'history';

const history = createBrowserHistory();

describe('post confirmation Error card page', () => {
  it('renders without crashing', () => {
    shallow(
      <Router history={history}>
        <PostConfirmError />
      </Router>
    );
  });

  it('renders and works correctly', () => {
    const props = {
      filename: 'defective.csv',
      errorObj: {
        errors: [
          {
            failureReason:
              'Product ID does not exist in product catalogue, Service point ID does not exist, Donor is not valid, PO Number should be a whole number',
            row: '1',
          },
          { failureReason: 'Service point ID does not exist', row: '2' },
          {
            failureReason:
              'Service point ID does not exist, UNICEF section is not valid, Donor is not valid',
            row: '3',
          },
          { failureReason: '', row: '' },
        ],
        rowsNumber: '3',
        rowsProcessed: '0',
      },
    };
    const wrapper = mount(
      <Router history={history}>
        <PostConfirmError {...props} />
      </Router>
    );

    // card title text
    expect(wrapper.find('.card-title').first().text()).toMatchSnapshot('card title');

    // fix csv instructions
    expect(toJson(wrapper.find('ol'))).toMatchSnapshot('fix csv instructions');

    // Table error page
    wrapper.find('tr').forEach((tr) => {
      expect(tr.text()).toMatchSnapshot('tr');
    });
  });
});
