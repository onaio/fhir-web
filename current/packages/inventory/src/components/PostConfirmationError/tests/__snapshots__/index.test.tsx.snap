// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post confirmation Error card page renders and works correctly: card title 1`] = `"Processing error: inventory items failed to be added"`;

exports[`post confirmation Error card page renders and works correctly: fix csv instructions 1`] = `
<ol>
  <li>
    Extract the rows listed below from "defective.csv"
  </li>
  <li>
    Paste the rows into a new CSV file
  </li>
  <li>
    <Link
      to="/inventory/upload"
    >
      <LinkAnchor
        href="/inventory/upload"
        navigate={[Function]}
      >
        <a
          href="/inventory/upload"
          onClick={[Function]}
        >
          Upload the CSV file
        </a>
      </LinkAnchor>
    </Link>
  </li>
</ol>
`;

exports[`post confirmation Error card page renders and works correctly: tr 1`] = `"Row numberErrors"`;

exports[`post confirmation Error card page renders and works correctly: tr 2`] = `"1Product ID does not exist in product catalogue, Service point ID does not exist, Donor is not valid, PO Number should be a whole number"`;

exports[`post confirmation Error card page renders and works correctly: tr 3`] = `"2Service point ID does not exist"`;

exports[`post confirmation Error card page renders and works correctly: tr 4`] = `"3Service point ID does not exist, UNICEF section is not valid, Donor is not valid"`;

exports[`post confirmation Error card page renders and works correctly: tr 5`] = `""`;
